File: ecos-records/src/main/java/ru/citeck/ecos/records3/graphql/meta/value/factory/LongValueFactory.java
package ru.citeck.ecos.records3.graphql.meta.value.factory;

import ru.citeck.ecos.records3.graphql.meta.value.MetaValue;

import java.util.Collections;
import java.util.List;

public class LongValueFactory implements MetaValueFactory<Long> {

    @Override
    public MetaValue getValue(Long value) {
        return new MetaValue() {
            @Override
            public String getString() {
                return value.toString();
            }

            @Override
            public Double getDouble() {
                return Double.valueOf(value);
            }

            @Override
            public Boolean getBool() {
                return value != 0;
            }
        };
    }

    @Override
    public List<Class<? extends Long>> getValueTypes() {
        return Collections.singletonList(Long.class);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/atts/value/factory/LongValueFactory.kt
package ru.citeck.ecos.records3.record.atts.value.factory

import ru.citeck.ecos.records3.record.atts.value.AttValue

class LongValueFactory : AttValueFactory<Long> {

    override fun getValue(value: Long): AttValue {

        return object : AttValue {

            override fun asText(): String {
                return value.toString()
            }

            override fun asDouble(): Double {
                return value.toDouble()
            }

            override fun asBoolean(): Boolean {
                return value != 0L
            }

            override fun asRaw(): Any {
                return value
            }
        }
    }

    override fun getValueTypes() = listOf(java.lang.Long::class.java)
}
